builtin/logical/ssh/path_sign.go

```diff
@@ -355,17 +460,24 @@ func (b *backend) calculateExtensions(data *framework.FieldData, role *sshRole)
 	return extensions, nil
 }
 
-func (b *backend) calculateTTL(data *framework.FieldData, role *sshRole) (time.Duration, error) {
+// return values:
+//  1. the calculated TTL duration (may be from specified data or derived from role/defaults)
+//  2. true if the returned TTL was specified in the data, false otherwise
+//  3. calculated maxTTL
+//  4. any error
+func (b *backend) calculateTTL(data *framework.FieldData, role *sshRole) (time.Duration, time.Duration, bool, error) {
```

Comment:
Modified the return signature here to return some pertinent information so the certificate sign function can make some new decisions:

1. whether `ttl` was specified or derived from roles/defaults (to be used to check if `ttl` was specified at the same time as `valid_after` and `valid_before`)
1. what the calculated `max_ttl` was (to be used to check if the period of time between `valid_after` and `valid_before` exceeds `max_ttl`